conda_build/metadata.py

```diff
@@ -269,9 +269,21 @@ def get_contents(meta_path):
     from conda_build.jinja_context import context_processor
 
     path, filename = os.path.split(meta_path)
-    loaders = [jinja2.PackageLoader('conda_build'),
+    loaders = [# search relative to '<conda_root>/Lib/site-packages/conda_build/templates'
+               jinja2.PackageLoader('conda_build'),
+               # search relative to RECIPE_DIR
                jinja2.FileSystemLoader(path)
                ]
+    
+    # search relative to current conda environment directory
+    conda_env_path = os.environ.get('CONDA_ENV_PATH',  # path to current conda environment
```

Comment:
It refers to the active environment where I run the build. 

Background: On Windows, binaries from different compiler versions tend to be incompatible. For example, when I use the official `libtiff` or `libpng` (created with Visual Studio 2008) in a build with Visual Studio 2012, the resulting program crashes. In order to support multiple compilers, I want to create environments like `vc9` (for Visual Studio 2008) and `vc11` (for Visual Studio 2012). Before each build, I activate the appropriate environment. Then `CONDA_ENVIRONMENT/toolset-info.yaml` contains all the necessary information to guarantee consistent builds.